nonsemantic/NonSemantic.ClspvReflection.asciidoc

```diff
@@ -0,0 +1,679 @@
+ +
+'Size' must be a 32-bit unsigned integer *OpConstant* specifying the size of the push 
+constant in bytes.
+1+| 18 | '<id>' +
+'Offset' |
+'<id>' +
+'Size'
+|=====
+
+[cols="3"]
+|=====
+3+| [[PushConstantNumWorkgroups]]*PushConstantNumWorkgroups* +
+ +
+Declares a *PushConstant* entry to specify the number of workgroups enqueued. All kernels from
+this module should include a push constant range that encompasses the 'Offset' and 'Size' operands. +
+ +
+'Offset' must be a 32-bit unsigned integer *OpConstant* specifying the offset in the block
+in bytes. +
+ +
+'Size' must be a 32-bit unsigned integer *OpConstant* specifying the size of the push 
+constant in bytes.
+1+| 19 | '<id>' +
+'Offset' |
+'<id>' +
+'Size'
+|=====
+
+[cols="3"]
+|=====
+3+| [[PushConstantRegionGroupOffset]]*PushConstantRegionGroupOffset* +
+ +
+Declares a *PushConstant* entry to specify the region group offset of a kernel. All kernels from
+this module should include a push constant range that encompasses the 'Offset' and 'Size' operands. +
+ +
+'Offset' must be a 32-bit unsigned integer *OpConstant* specifying the offset in the block
+in bytes. +
+ +
+'Size' must be a 32-bit unsigned integer *OpConstant* specifying the size of the push 
+constant in bytes.
+1+| 20 | '<id>' +
+'Offset' |
+'<id>' +
+'Size'
+|=====
+
+[cols="4"]
+|=====
+4+| [[ConstantDataStorageBuffer]]*ConstantDataStorageBuffer* +
+Declares a storage buffer to hold constant data specified by 'Data'. All kernels from this module
+should include a descriptor with the type *VK_DESCRIPTOR_TYPE_STORAGE_BUFFER* that is backed by
+a buffer initialized with 'Data'. +
+ +
+'DescriptorSet' must be a 32-bit unsigned integer *OpConstant* specifying the descriptor set. +
+ +
+'Binding' must be a 32-bit unsigned integer *OpConstant* specifying the binding. +
+ +
+'Data' must be an *OpString* that encodes the hexbytes of the constant data.
+1+| 21 | '<id>' +
+'DescriptorSet' |
+'<id>' +
+'Binding' |
+'<id>' +
+'Data'
+|=====
+
+[cols="4"]
+|=====
+4+| [[ConstantDataUniform]]*ConstantDataUniform* +
+Declares a uniform buffer to hold constant data specified by 'Data'. All kernels from this module
+should include a descriptor with the type *VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER* that is backed by
+a buffer initialized with 'Data'. +
+ +
+'DescriptorSet' must be a 32-bit unsigned integer *OpConstant* specifying the descriptor set. +
+ +
+'Binding' must be a 32-bit unsigned integer *OpConstant* specifying the binding. +
+ +
+'Data' must be an *OpString* that encodes the hexbytes of the constant data.
+1+| 22 | '<id>' +
+'DescriptorSet' |
+'<id>' +
+'Binding' |
+'<id>' +
+'Data'
+|=====
+
+[cols="4"]
+|=====
+4+| [[LiteralSampler]]*LiteralSampler* +
+Declares a literal sampler used by the module. All kernels from this module should include a
+descriptor with the type *VK_DESCRIPTOR_TYPE_SAMPLER* that has the properties encoded by 'Mask'
+(see https://github.com/google/clspv/blob/master/include/clspv/Sampler.h[Sampler.h]). +
```

Comment:
This link is broken.

Reply:
This should be the location when this PR is merged.